server/src/test/java/org/elasticsearch/index/mapper/flattened/RootFlattenedFieldTypeTests.java

```diff
@@ -135,9 +160,61 @@ public void testWildcardQuery() {
 
     public void testFetchSourceValue() throws IOException {
         Map<String, Object> sourceValue = Map.of("key", "value");
-        RootFlattenedFieldType ft = createDefaultFieldType();
+        RootFlattenedFieldType ft = createDefaultFieldType(Integer.MAX_VALUE);
 
         assertEquals(List.of(sourceValue), fetchSourceValue(ft, sourceValue));
         assertEquals(List.of(), fetchSourceValue(ft, null));
     }
+
+    public void testFetchSourceValueWithIgnoreAbove() throws IOException {
+        Map<String, Object> sourceValue = Map.of("key", "test ignore above");
+
+        assertEquals(List.of(), fetchSourceValue(createDefaultFieldType(10), sourceValue));
+        assertEquals(List.of(sourceValue), fetchSourceValue(createDefaultFieldType(20), sourceValue));
+    }
+
+    public void testFetchSourceValueWithList() throws IOException {
+        Map<String, Object> sourceValue = Map.of("key1", List.of("one", "two", "three"));
+
+        assertEquals(List.of(Map.of("key1", List.of("one", "two"))), fetchSourceValue(createDefaultFieldType(3), sourceValue));
+    }
+
+    public void testFetchSourceValueWithMultipleFields() throws IOException {
+        Map<String, Object> sourceValue = Map.of(
+            "key1",
+            "test",
+            "key2",
+            List.of("one", "two", "three"),
+            "key3",
+            "hi",
+            "key4",
+            List.of("the quick brown fox", "jumps over the lazy dog")
```

Comment:
Can we add `"key5", Map.of("nested", List.of("foo", "foobar")`?

We can have subobjects, iiuc.

Reply:
I don't think so I would expect a case like the one you described to be handled like

```
"key5.nested" -> ["foo", "bar"] because that is how flattening works, right?
```

Reply:
Yeah makes sense.. Let's add this to yaml for peace of mind, to verify the expected behavior.